packages/react-dom/src/events/ReactDOMEventReplaying.js

```diff
@@ -532,6 +543,8 @@ function replayUnblockedEvents() {
           nextDiscreteEvent.nativeEvent,
         );
         if (nextBlockedOn === null) {
+          // This whole function is in !enableCapturePhaseSelectiveHydrationWithoutDiscreteEventReplay,
+          // so we don't need the new replay behavior code branch.
```

Comment:
I don't like that I'm able to remove the contents of this whole if statement and none of the tests seem to fail, even if I hardcode `enableClientRenderFallbackOnHydrationMismatch` in `ReactFeatureFlags.www-dynamic` to `false`.

It would be nice to have a test verifying this does something. Though it is old behavior, I feel uncomfortable it's not tested until it's deleted.

Reply:
Weird, there should be coverage for this block I'll double check.

Reply:
Hmm, I think I'm just going to delete it now since we've been testing with discrete event replay off for a while so I'll put up a PR for that.